contrib/win32/openssh/OpenSSHUtils.psm1

```diff
@@ -0,0 +1,410 @@
+}
+
+<#
+    .Synopsis
+    Fix-AuthorizedKeyPermissions
+    fix the file owner and permissions of authorized_keys
+#>
+function Fix-AuthorizedKeyPermissions
+{
+    param (
+        [parameter(Mandatory=$true)]
+        [ValidateNotNullOrEmpty()]
+        [ValidateScript({Test-Path $_ })]
+        [string]$FilePath,
+        [switch] $Quiet)
+        
+        $fullPath = (Resolve-Path $FilePath).Path
+        $profileListPath = "HKLM:\SOFTWARE\Microsoft\Windows NT\CurrentVersion\ProfileList"
+        $profileItem = Get-ChildItem $profileListPath  -ErrorAction Ignore | ? { 
+            $fullPath.ToLower().Contains((Get-ItemPropertyValue $_.PSPath -Name ProfileImagePath -ErrorAction Ignore).Tolower())
+        }
+        $userSid = $profileItem.PSChildName
+        $account = Get-UserSID -UserSid $userSid
+        Fix-FilePermissions -Owners $account,$adminsAccount,$systemAccount -AnyAccessOK $account -ReadAccessNeeded $sshdAccount @psBoundParameters
+}
+
+<#
+    .Synopsis
+    Fix-UserKeyPermissions
+    fix the file owner and permissions of user config
+#>
+function Fix-UserKeyPermissions
+{
+    param (
+        [parameter(Mandatory=$true)]
+        [ValidateNotNullOrEmpty()]
+        [ValidateScript({Test-Path $_ })]
+        [string]$FilePath,
+        [switch] $Quiet)
+
+        Fix-FilePermissions -Owners $currentUser, $adminsAccount,$systemAccount -AnyAccessOK $currentUser @psBoundParameters
+        $publicParameters = $PSBoundParameters
+        $publicParameters["FilePath"] += ".pub"
+        Fix-FilePermissions -Owners $currentUser, $adminsAccount,$systemAccount -AnyAccessOK $currentUser -ReadAccessOK $everyone @publicParameters
+}
+
+<#
+    .Synopsis
+    Fix-UserSSHConfigPermissions
+    fix the file owner and permissions of user config
+#>
+function Fix-UserSSHConfigPermissions
+{
+    param (
+        [parameter(Mandatory=$true)]
+        [ValidateNotNullOrEmpty()]
+        [ValidateScript({Test-Path $_ })]
+        [string]$FilePath,
+        [switch] $Quiet)
+        Fix-FilePermissions -Owners $currentUser,$adminsAccount,$systemAccount -AnyAccessOK $currentUser @psBoundParameters
+}
+
+<#
+    .Synopsis
+    Fix-FilePermissionInternal
+    Only validate owner and ACEs of the file
+#>
+function Fix-FilePermissions
+{
+    param (        
+        [parameter(Mandatory=$true)]
+        [ValidateNotNullOrEmpty()]
+        [ValidateScript({Test-Path $_ })]
+        [string]$FilePath,
+        [ValidateNotNull()]
+        [System.Security.Principal.NTAccount[]] $Owners = $currentUser,
+        [System.Security.Principal.NTAccount[]] $AnyAccessOK,
+        [System.Security.Principal.NTAccount[]] $ReadAccessOK,
+        [System.Security.Principal.NTAccount[]] $ReadAccessNeeded,
+        [switch] $Quiet
+    )   
+    
+    Write-host "----------Validating $FilePath----------"
+    $return = Fix-FilePermissionInternal @PSBoundParameters
+
+    if($return -contains $true) 
```

Comment:
it will return true when we need to call the function again; otherwise, we don't need to call the function again